docs/apm-mutating-webhook.asciidoc

```diff
@@ -0,0 +1,230 @@
+instrumented by an Elastic APM agent, and a <<apm-helm-chart,Helm chart>> that manages the webhook
+receiver's lifecycle within Kubernetes.
+
+Learn more below, or skip ahead to <<apm-get-started-webhook>>.
+
+[[apm-webhook]]
+=== Webhook
+
+The webhook receiver modifies pods so they are automatically instrumented by an Elastic APM agent.
+Supported agents include:
+
+// links will be added later
+* Java agent
+* .NET agent
+* Node.js agent
+
+The webhook receiver is invoked on pod creation. After receiving the object definition from the Kubernetes
+API server, it looks through the pod spec for a specific, user-supplied annotation. If found, the pod spec
+is mutated according to the webhook receiver's configuration. This mutated object is then returned to the
+Kubernetes API server which uses it as the source of truth for the object.
+
+[[apm-mutation]]
+==== Mutation
+
+The mutation that occurs is defined below:
+
+. Add an init container image that has the agent binary.
+. Add a shared volume that is mounted into both the init container image and
+all container images contained in the original incoming object.
+. Copy the agent binary from the init container image into the shared volume,
+making it available to the other container images.
+. Update the environment variables in the container images to configure
+auto-instrumentation with the copied agent binary
+
+TIP: To learn more about mutating webhooks,
+see the {kube-admin-docs}[Kubernetes Admission controller documentation].
+
+[[apm-helm-chart]]
+=== Helm chart
+
+The Helm chart manages the configuration of all associated manifest files for the
+webhook receiver, including generating certificates for securing communication
+between the Kubernetes API server and the webhook receiver.
+
+TIP: To learn more about Helm charts, see the {helm-docs}[Helm documentation].
+
+// Break content below to a new page
+
+[[apm-get-started-webhook]]
+== Get started with the webhook
+
+To instrument and configure your application pods, complete the following steps:
+
+. <<apm-webhook-clone-repo>>
+. <<apm-webhook-configure-helm>>
+. <<apm-webhook-install-helm>>
+. <<apm-webhook-add-pod-annotation>>
+// Anything next? Restart the deployment maybe?
+
+[float]
+[[apm-webhook-clone-repo]]
+=== Clone the repository
+
+Clone the `elastic/apm-mutating-webhook` repository:
+
+[source,bash]
+----
+git clone git@github.com:elastic/apm-mutating-webhook.git
+cd apm-mutating-webhook
+----
+
+[float]
+[[apm-webhook-configure-helm]]
+=== Configure the webhook with a Helm values file
+
+The webhook is installed with a Helm chart.
+You can provide a custom webhook configuration using a Helm values file.
+Elastic provides a https://github.com/elastic/apm-mutating-webhook/blob/main/custom.yaml[`custom.yaml`] file as a starting point.
+
+Here's an example of a `custom.yaml` file that instruments a pod with the Elastic APM Java agent:
```

Comment:
If we choose to keep the agent instructions in one place for the technical preview, it probably makes sense to have an example per agent. Otherwise, it might be difficult for users to determine, for example, the docker `image` of the agent.

Reply:
+1 on each language getting a sample configuration here.

Reply:
You don't happen to have a sample `custom.yaml` file for Node.js do ya?